src/item.cpp

```diff
@@ -180,14 +184,10 @@ item item::make_corpse( const mtype_id& mt, int turn, const std::string &name )
         debugmsg( "tried to make a corpse with an invalid mtype id" );
     }
 
-    item result( "corpse", turn >= 0 ? turn : int( calendar::turn ) );
+    item result( mt.obj().get_corpse_itype(), turn >= 0 ? turn : int( calendar::turn ) );
```

Comment:
`string_id<>` overloads the arrow operator. I'd use it instead of `obj()` for the sake of readability: `mt->get_corpse_itype()`